src/metabase/db/liquibase.clj

```diff
@@ -118,6 +100,34 @@
   ^Boolean [^Liquibase liquibase]
   (boolean (seq (.listLocks liquibase))))
 
+(defn force-release-locks!
+  "(Attempt to) force release Liquibase migration locks."
+  [^Liquibase liquibase]
+  (.forceReleaseLocks liquibase))
+
+(defn release-lock-if-needed!
+  "Attempts to release the liquibase lock if present. Logs but does not bubble up the exception if one occurs as it's
+  intended to be used when a failure has occurred and bubbling up this exception would hide the real exception."
+  [^Liquibase liquibase]
+  (when (migration-lock-exists? liquibase)
+    (try
+      (force-release-locks! liquibase)
+      (catch Exception e
+        (log/error e (trs "Unable to release the Liquibase lock after a migration failure"))))))
+
+(defn migrations-sql
+  "Return a string of SQL containing the DDL statements needed to perform unrun `liquibase` migrations.
+  Note that custom migrations are not included."
+  ^String [^Liquibase liquibase]
+  ;; calling update on custom migrations will execute them, and we don't want it to happen
+  ;; so ignore its
+  (doseq [^ChangeSet change (.listUnrunChangeSets liquibase nil nil)]
```

Comment:
without doing this, whenever calling `migrations-sql` all the custom migrations will be executed, but not the SQL ones